src/data.ml

```diff
@@ -86,14 +86,12 @@ module Entry = struct
     let decode_value string off = V.decode string (off + K.encoded_size)
 
     let encode' key value f =
-      let encoded_key = K.encode key in
-      let encoded_value = V.encode value in
-      if String.length encoded_key <> K.encoded_size then
-        raise (Invalid_size encoded_key);
-      if String.length encoded_value <> V.encoded_size then
-        raise (Invalid_size encoded_value);
-      f encoded_key;
-      f encoded_value
```

Comment:
By the way I suspect this was actually a bug, as `f` here is usually `IO.append`, which can flush. 
So a `IO.flush` can basically be called between the key and value, and I think this would be an inconsistent state, since we assume in multiple places that `offset` is a multiple of `entry_size` for instance?
/cc @icristescu @CraigFe

Reply:
Agreed. We should be more careful about auto-flushing in the middle of "transaction"s in future.

Reply:
I believe this happens in multiple other places too... I like the `append_transaction` API endpoint.